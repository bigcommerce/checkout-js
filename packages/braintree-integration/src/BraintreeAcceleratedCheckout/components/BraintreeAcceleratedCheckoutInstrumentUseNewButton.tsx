import { noop } from 'lodash';
import React, { FunctionComponent } from 'react';

import { TranslatedString } from '@bigcommerce/checkout/locale';
import { CreditCardIcon } from '@bigcommerce/checkout/ui';

interface BraintreeAcceleratedCheckoutInstrumentUseNewButtonProps {
    className?: string;
    onClick?(): void;
}

const BraintreeAcceleratedCheckoutInstrumentUseNewButton: FunctionComponent<
    BraintreeAcceleratedCheckoutInstrumentUseNewButtonProps
> = ({ className, onClick = noop }) => (
    <button
        className={className}
        data-test="instrument-select-option-use-new"
        onClick={onClick}
        type="button"
    >
        <div className="instrumentSelect-details instrumentSelect-details--addNew">
            <CreditCardIcon />

            <div className="instrumentSelect-card">
                <TranslatedString id="payment.instrument_add_card_action" />
            </div>
        </div>
    </button>
);

export default BraintreeAcceleratedCheckoutInstrumentUseNewButton;
