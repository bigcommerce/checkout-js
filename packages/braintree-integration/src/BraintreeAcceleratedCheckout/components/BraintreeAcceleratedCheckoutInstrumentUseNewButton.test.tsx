import { render } from '@testing-library/react';
import React from 'react';

import BraintreeAcceleratedCheckoutInstrumentUseNewButton from './BraintreeAcceleratedCheckoutInstrumentUseNewButton';

describe('BraintreeAcceleratedCheckoutInstrumentUseNewButton', () => {
    it('renders "new vaulted instrument" button', () => {
        const view = render(<BraintreeAcceleratedCheckoutInstrumentUseNewButton />);

        expect(view).toMatchSnapshot();
    });
});
