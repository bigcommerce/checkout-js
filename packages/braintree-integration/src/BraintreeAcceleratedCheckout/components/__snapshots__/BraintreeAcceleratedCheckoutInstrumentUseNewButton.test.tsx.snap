// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BraintreeAcceleratedCheckoutInstrumentUseNewButton renders "new vaulted instrument" button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        data-test="instrument-select-option-use-new"
        type="button"
      >
        <div
          class="instrumentSelect-details instrumentSelect-details--addNew"
        >
          <div
            class="cardIcon-icon cardIcon-icon--default icon icon--medium"
          />
          <div
            class="instrumentSelect-card"
          />
        </div>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      data-test="instrument-select-option-use-new"
      type="button"
    >
      <div
        class="instrumentSelect-details instrumentSelect-details--addNew"
      >
        <div
          class="cardIcon-icon cardIcon-icon--default icon icon--medium"
        />
        <div
          class="instrumentSelect-card"
        />
      </div>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
